libraries/AP_Notify/Buzzer.cpp

```diff
@@ -207,6 +204,10 @@ void Buzzer::update()
 // on - turns the buzzer on or off
 void Buzzer::on(bool turn_on)
 {
+    if (!pNotify->buzzer_enabled())	{
+		turn_on = false;
```

Comment:
I'm not sure I like this here, it will still be executing all the `play_pattern` and `on` calls on the `update` method. I think it would be nicer if this is was the first check in the `update` method; if buzzer wasn't enabled, then call `on(false)` and return immediately.